Fullscript/js-transforms
# Added aviaryMigrator transform for converting modifying component names, import location, props etc.

## Purpose

This new transform primarily is to be used for the aviary design system. It can be used to convert/modify packages as part of a large migration. For example changing the import location of a component, renaming some of its props, deleting props, adding new props with values as well as rename the component itself.

## Testing

1. Pull down this codebase and PR locally by cloning and doing a `git fetch origin`
2. Change to this branch
3. install dependencies `yarn install`
4. Run `yarn link` to create link to your local package
5. In the repo you want to test, run `yarn link @fullscript/js-transforms`
6. Create a configuration file in the repo you want to test like this:

```json
{
  "componentName": "Status",
  "importSource": "@aviary",
  "newImportSource": "@fullscript/aviary-web",
  "propsToRemove": ["dotPlacement"],
  "propsToRename": {
    "css": "customCss",
    "isColor": "intention"
  },
  "propsToAdd": {
    "intention": "success"
  }
}
```

7. Run the transform with the following command: `yarn js-transforms aviaryMigrator './path/to/files/**/*.ts*' --options ./PathToConfig.json --verbose`
8. Make sure to run eslint format and prettier format
9. Inspect the changes, do they make sense?

## Future

Additional things todo that can be done as a follow up: 
- Support namespaced components such as Columns.Column
- Support migrating multiple components at once e.g Columns and Column
- A nice interactive CLI with declarative options/prompts
